inheritance/src/main/java/inheritance/Restaurant.java

```diff
@@ -0,0 +1,82 @@
+package inheritance;
+
+// Thanks to James Dansie for providing help with the logic and some code
+
+// Restaurant gets a linked list of reviews.
+// The methods exist to update the constructor function, given a review added.
+
+public class Restaurant {
+    String name;
+    double stars0To5;
+    int priceCategory;
+
+    // constructor for Restaurant
+    public Restaurant (String name, double stars0To5, int priceCategory) {
+        this.name = name;
+        this.priceCategory = priceCategory;
+        this.stars0To5 = stars0To5;
+    }
+
+    // creates a linked list
+    Node head;
+
+    // Each node contains review body, review author and review star rating, each of which has to be
+    // added to its respective node. First, create the class Node, establish a method to add a node to the list, then create
+    // methods to add the data (body, author, star) to its respective node.
+
+    class Node {
+        Review data;
+        Node next;
+
+        public Node (Review data, Node next){
+            this.data = data;
+            this.next = next;
+        }
```

Comment:
This Node class is unexpected.. where you wanting to use a linkedlist? If so, you can just use the java.util.linkedlist. You don't have to make one from scratch.